api/src/utilities/passport-validator-utilities.ts

```diff
@@ -0,0 +1,71 @@
+import { HttpException, HttpStatus } from '@nestjs/common';
+
+/**
+ *
+ * @param lastLoginAt the last time the user logged in (stored in db)
+ * @param failedLoginAttemptsCount the number of times the user failed to log in (stored in db)
+ * @param maxAttempts the maximum number of attempts before user is considered locked out (env variable)
+ *
+ * @returns throws error if user is already locked out
+ */
+export function isUserLockedOut(
+  lastLoginAt: Date,
+  failedLoginAttemptsCount: number,
+  maxAttempts: number,
+  cooldown: number,
```

Comment:
Do you anticipate the maxAttempts and cooldown numbers to change per request type? We could just grab the env variables in this method instead of passing them in. But if you want it to be more versatile then I'm cool with this as well

Reply:
yeah I think its best to keep this versatile as just a helper function